osu.Game.Tests/Visual/Multiplayer/TestSceneMatchStartControl.cs

```diff
@@ -378,6 +378,31 @@ public void TestManyUsersChangingState(bool isHost)
             }, users);
         }
 
+        [Test]
+        public void TestAbortMatch()
+        {
+            multiplayerClient.Setup(m => m.StartMatch())
+                             .Callback(() =>
+                             {
+                                 multiplayerClient.Raise(m => m.LoadRequested -= null);
+                                 multiplayerClient.Object.Room!.State = MultiplayerRoomState.WaitingForLoad;
+
+                                 // The local user state doesn't really matter, so let's do the same as the base implementation for these tests.
+                                 changeUserState(localUser.UserID, MultiplayerUserState.Idle);
+                             });
+
+            multiplayerClient.Setup(m => m.AbortMatch())
+                             .Callback(() =>
+                             {
+                                 multiplayerClient.Object.Room!.State = MultiplayerRoomState.Open;
+                                 raiseRoomUpdated();
+                             });
+
+            ClickButtonWhenEnabled<MultiplayerReadyButton>();
+            ClickButtonWhenEnabled<MultiplayerReadyButton>();
+            ClickButtonWhenEnabled<MultiplayerReadyButton>();
```

Comment:
This is a bit wtf on the face of it (because it's ready up -> start match -> abort). May be worth adding inline comments for explanation.

Also, no asserts at the end of this?